### PR TITLE
[Android] Change the library load logic for embedded application.

### PR DESCRIPTION
**Pull request by wuhengzhi**
For embedded application, it loads the library from "app-lib" at first,
if that failed, then loads library from "/data/data/". This will also be
failed when the architecture is inconsistent and lead to crash.
So architecture should be checked before library loaded from "/data/data/".

BUG=XWALK-5804

### CODE REVIEW

**Comment by wuhengzhi** [comment]
@axinging @fujunwei @wuhengzhi @xzhan96 @XiaosongWei PTAL, thanks.

**Comment by wuhengzhi** [comment]
This PR should be back ported to Crosswalk-16.

**Comment by darktears** [comment]
Can you rephrase the commit there is some mistakes in english. Also it would be nice to add a test.

**Comment by xzhan96** [comment]
yeah, I also think it better to rephrase the commit log.

**Comment by wuhengzhi** [comment]
Please don't merge this PR.
